cmd/glasskube/cmd/update.go

```diff
@@ -29,24 +33,45 @@ var updateCmd = &cobra.Command{
 		updater := update.NewUpdater(client).
 			WithStatusWriter(statuswriter.Spinner())
 
-		tx, err := updater.Prepare(ctx, packageNames)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "❌ update preparation failed: %v\n", err)
-			os.Exit(1)
-		}
+		if len(args) == 1 && updateCmdOptions.Version != "" {
+			tx, err := updater.VersionUpdate(ctx, args[0], updateCmdOptions.Version)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error in updating the package version : %v", err)
+				os.Exit(1)
+			}
 
-		printTransaction(*tx)
+			printTransaction(*tx)
 
-		if !tx.IsEmpty() {
-			if !cliutils.YesNoPrompt("Do you want to apply these updates?", false) {
-				fmt.Fprintf(os.Stderr, "⛔ Update cancelled. No changes were made.\n")
-				os.Exit(0)
+			if !tx.IsEmpty() {
+				if !cliutils.YesNoPrompt("Do you want to apply these updates?", false) {
+					fmt.Fprintf(os.Stderr, "⛔ Update cancelled. No changes were made.\n")
+					os.Exit(0)
+				}
+				if err := updater.Apply(ctx, tx); err != nil {
+					fmt.Fprintf(os.Stderr, "❌ update failed: %v\n", err)
+					os.Exit(1)
+				}
 			}
-
-			if err := updater.Apply(ctx, tx); err != nil {
-				fmt.Fprintf(os.Stderr, "❌ update failed: %v\n", err)
+		} else {
+			tx, err := updater.Prepare(ctx, packageNames)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "❌ update preparation failed: %v\n", err)
 				os.Exit(1)
 			}
+
+			printTransaction(*tx)
+
+			if !tx.IsEmpty() {
+				if !cliutils.YesNoPrompt("Do you want to apply these updates?", false) {
+					fmt.Fprintf(os.Stderr, "⛔ Update cancelled. No changes were made.\n")
+					os.Exit(0)
+				}
+
+				if err := updater.Apply(ctx, tx); err != nil {
+					fmt.Fprintf(os.Stderr, "❌ update failed: %v\n", err)
+					os.Exit(1)
+				}
+			}
```

Comment:
this whole part does not need to be duplicated, it's the same for both decision branches.